core/arch/riscv64/mangle.c

```diff
@@ -955,9 +966,13 @@ mangle_exclusive_load(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
     ASSERT(opnd_is_reg(dst) && opnd_is_base_disp(src0));
     if (opnd_get_reg(dst) == dr_reg_stolen) {
         opnd_replace_reg(&dst, dr_reg_stolen, scratch_reg2);
+    } else if (opnd_get_reg(dst) == DR_REG_TP) {
+        opnd_replace_reg(&dst, DR_REG_TP, scratch_reg3);
```

Comment:
Could we use instr_replace_reg_resize() which will cover all operands for us?
Then we have just two calls to that, one for dr_reg_stolen and one for DR_REG_TP?

Reply:
Thanks for your hint! But currently riscv implementation of `opnd_replace_reg_resize()` is a stub[1], thus `instr_replace_reg_resize()` doesn't work. I'm willing to defer this change until the stub is replaced with a real implementation.

[1]: https://github.com/DynamoRIO/dynamorio/blob/4e6e71d337a4dcf46d7361d03c0ba0b354759a39/core/ir/opnd_shared.c#L1586-L1600

Reply:
Or add a XXX comment here to replace when that is available for simpler code and go ahead and merge this PR.

Reply:
> Or add a XXX comment here to replace when that is available for simpler code and go ahead and merge this PR.

Done